lib/msal-browser/src/cache/BrowserStorage.ts

```diff
@@ -427,4 +401,65 @@ export class BrowserStorage extends CacheManager {
             throw BrowserAuthError.createTokenRequestCacheError(err);
         }
     }
+
+    /**
+     * helper to retrieve the value with type from the cache
+     * @param key
+     * @param value
+     */
+    private getItemHelper(key: string, value: string): ValidCacheType {
+
+        // handle null cases
+        if (StringUtils.isEmpty(value)) {
+            return null;
```

Comment:
If the value from browser storage is empty, we should not immediately return null, as it doesnt give the library the opportunity to see if the value is stored in a cookie. Instead, if `value` is `null` from local/session storage and `storeAuthStateInCookie` is true, then check for the value stored in a cookie.